ethereum/core/src/main/java/org/hyperledger/besu/ethereum/privacy/FlexibleUtil.java

```diff
@@ -40,11 +42,53 @@ public static boolean isGroupAdditionTransaction(final PrivateTransaction privat
   }
 
   public static List<String> getParticipantsFromParameter(final Bytes input) {
+    final int numberOfParticipants;
+    try {
+      numberOfParticipants = input.slice(4 + 32, 32).toBigInteger().intValue();
+    } catch (final Exception exception) {
+      return Collections.emptyList();
+    }
+    if (numberOfParticipants == 0) return Collections.emptyList();
+    // Method selector + offset +  number of participants + (offset * number of participants)
+    final Bytes mungedParticipants = input.slice(4 + 32 + 32 + (32 * numberOfParticipants));
+
+    return getParticipantsFromMungedParticipants(mungedParticipants, numberOfParticipants);
+  }
+
+  public static List<String> decodeList(final Bytes rlpEncodedList) {
+    // first 32 bytes is dynamic list offset
+    if (rlpEncodedList.size() < 64) return new ArrayList<>();
+    // Bytes uses a byte[] for the content which can only have up to Integer.MAX_VALUE-5 elements
+    final int lengthOfList =
+        UInt256.fromBytes(rlpEncodedList.slice(32, 32)).toInt(); // length of list
+    if (lengthOfList == 0 || rlpEncodedList.size() < 64 + lengthOfList * 32) {
+      return new ArrayList<>();
+    }
+
+    final Bytes mungedParticipants = rlpEncodedList.slice(32 + 32 + (32 * lengthOfList));
```

Comment:
can we think of a better name than `mungedParticipants`?

Reply:
Changed to `encodedParticipants`